### PR TITLE
feat: user restriction rework

### PR DESCRIPTION
**Pull request by or2e**
**Patch notes:**

## Webui
- Blocking reason added. @or2e 
  <details>
    <summary>Screenshots</summary>

    image
  </details>

___
## Moderators only
- For restrictions created now there are 2 fields - public and private reason for blocking. Obviously, the public one will be seen by the user. @or2e 
  <details>
    <summary>Screenshots</summary>

    image
    image
  </details>

### CODE REVIEW

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
```
Comment (namidaka) [inline src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs:4]: remove

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+namespace Crpg.Persistence.Migrations
+{
```
Comment (namidaka) [inline src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs:6]: scoped namespace namespace Crpg.Persistence.Migrations;

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs [hunk 3]
```diff
@@ -0,0 +1,2182 @@
...
+using Crpg.Persistence;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+using NetTopologySuite.Geometries;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+
+#nullable disable
+
```
Comment (namidaka) [inline src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs:22]: remove

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs [hunk 4]
```diff
@@ -0,0 +1,2182 @@
...
+namespace Crpg.Persistence.Migrations
+{
```
Comment (namidaka) [inline src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs:24]: Scoped namespace namespace Crpg.Persistence.Migrations;

**Review by namidaka — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs (hunk 2 above)
Comment (or2e) [reply src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs:6]: +

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs (hunk 3 above)
Comment (or2e) [reply src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs]: +

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs (hunk 1 above)
Comment (or2e) [reply src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.cs]: ++

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs (hunk 4 above)
Comment (namidaka) [reply src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs:22]: Still not scoped?

**Review by namidaka — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs (hunk 4 above)
Comment (or2e) [reply src/Persistence/Migrations/20240103030850_AddRestrictionPublicReason.Designer.cs:22]: fix fix fix )

**Review by namidaka — APPROVED** [review]

**Pull request merged**